clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/ops/UpsertAzureAppGatewayAtomicOperation.groovy

```diff
@@ -96,51 +97,81 @@ class UpsertAzureAppGatewayAtomicOperation implements AtomicOperation<Map> {
         errList = AzureDeploymentOperation.checkDeploymentOperationStatus(task, BASE_PHASE, description.credentials, resourceGroupName, deployment.name)
       } else {
         // We are attempting to create a new application gateway
-        description.credentials.resourceManagerClient.initializeResourceGroupAndVNet(description.credentials, resourceGroupName, virtualNetworkName, description.region)
+        if (!description.useDefaultVnet) {
+          task.updateStatus(BASE_PHASE, "Create ApplicationGateway using virtual network $description.vnet and subnet $description.subnet for server group $description.name")
 
-        task.updateStatus(BASE_PHASE, "Creating subnet for application gateway")
+          // Create corresponding ResourceGroup if it's not created already
+          description.credentials.resourceManagerClient.initializeResourceGroupAndVNet(description.credentials, resourceGroupName, null, description.region)
```

Comment:
If you pass null for the virtual network name then it's the same as if you passed _virtualNetworkName_ since passing null will cause this method to call AzureUtilities.getVirtualNetworkName(resourceGroupName). So in effect, this line is the same as the corresponding line in the *else* clause below (line 126). Is that what you were intending?